plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -154,53 +159,58 @@ private void checkInputReady()
 
     private synchronized List<ExchangeSourceHandle> createExchangeSourceHandles()
     {
-        Multimap<Integer, String> partitionFiles = ArrayListMultimap.create();
-        for (LocalFileSystemExchangeSinkHandle sinkHandle : finishedSinks) {
-            Path committedAttemptPath = getCommittedAttemptPath(sinkHandle);
-            Map<Integer, Path> partitions = getCommittedPartitions(committedAttemptPath);
-            partitions.forEach((partition, file) -> partitionFiles.put(partition, file.toAbsolutePath().toString()));
+        Multimap<Integer, URI> partitionFiles = ArrayListMultimap.create();
```

Comment:
overloading `partitions` for source and output makes it somewhat hard to read.
Maybe raname `finishedPartitions` to `finishedTasks` or `finishedTaskPartitions` and `paritionsId` to `task` or `taskPartition`

Reply:
Decided to go with `finishedSinks` and `taskPartition`